### ISSUE TITLE
Expressing dependencies via self-types

### ISSUE THREAD

**Issue by backuitist** [di-in-scala/di-in-scala.github.io-sources#3]
The example in this section does not exhibit any self-type, is that right?

**Comment by adamw** [comment]
True, it only uses the "or extending other trait-modules" :). And now self-types even work!

**Comment by backuitist** [comment]
Shit, wrong button!
Maybe the section could then be renamed into "Expressing dependencies by extending other trait-modules"? There could also be another section for self-types?

**Comment by adamw** [comment]
Yeah I'll fix that when updating docs when 2.x is out.

As most of the code is the same, I though about extending the examples to show both variants.
